backend/src/Wasm/Libs/WASM.fs

```diff
@@ -13,8 +13,186 @@ open LibExecution.StdLib.Shortcuts
 module PT = LibExecution.ProgramTypes
 module Exe = LibExecution.Execution
 
+open System.IO
+open System.Net.Http
+
+open System.Threading.Tasks
+open FSharp.Control.Tasks
+
+open LibExecution
+open VendoredTablecloth
+
+
+module HttpClient =
+  type Method = HttpMethod
+
+  module Headers =
+    type Header = string * string
+    type T = List<Header>
+
+  type Body = byte array
+
+  type HttpRequest =
+    { url : string
+      method : Method
+      headers : Headers.T
+      body : Body }
+
+  type HttpResult = { statusCode : int; headers : Headers.T; body : Body }
+
+  // forked from Elm's HttpError type
+  // https://package.elm-lang.org/packages/elm/http/latest/Http#Error
+
+  type HttpRequestError =
+    | BadUrl of details : string
+    | Timeout
+    | NetworkError
+    | Other of details : string
+
+  type HttpRequestResult = Result<HttpResult, HttpRequestError>
+
+
+  let private httpClient : HttpClient =
+    new HttpClient(
+      new HttpClientHandler(AllowAutoRedirect = false),
+      disposeHandler = false,
+      Timeout = System.TimeSpan.FromSeconds 30.0,
+      MaxResponseContentBufferSize = 1024L * 1024L * 100L // 100MB
+    )
+
+  let request (httpRequest : HttpRequest) : Task<HttpRequestResult> =
+    task {
+      // use _ =
+      // Telemetry.child
+      //   "HttpClient.call"
+      //   [ "request.url", httpRequest.url; "request.method", httpRequest.method ]
+      try
+        let uri = System.Uri(httpRequest.url, System.UriKind.Absolute)
+
+        // currently we only support http(s) requests
+        if uri.Scheme <> "https" && uri.Scheme <> "http" then
+          return Error(BadUrl "Unsupported Protocol")
+        else
+          let reqUri =
+            System.UriBuilder(
+              Scheme = uri.Scheme,
+              Host = uri.Host,
+              Port = uri.Port,
+              Path = uri.AbsolutePath,
+              Query = uri.Query
+            )
+            |> string
+
+          use req =
+            new HttpRequestMessage(
+              httpRequest.method,
+              reqUri,
+              // Content = new ByteArrayContent(httpRequest.body),
```

Comment:
To check:
Uncommenting Content, I get a TypeError
ypeError: Failed to execute 'fetch' on 'Window': Request with GET/HEAD method cannot have body. TypeError: Failed to execute 'fetch' on 'Window': Request with GET/HEAD method cannot have body.